service/service.go

```diff
@@ -1,49 +1,70 @@
 package service
 
 import (
+	"errors"
 	"fmt"
 	"os"
 	"os/signal"
+	"sync/atomic"
+	"syscall"
+	"time"
 
+	"golang.org/x/net/context"
+
+	"github.com/ONSdigital/dp-dimension-extractor/api"
 	"github.com/ONSdigital/go-ns/kafka"
 	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/rchttp"
 	"github.com/ONSdigital/go-ns/s3"
 )
 
 // Service represents the necessary config for dp-dimension-extractor
 type Service struct {
-	EnvMax              int64
-	Consumer            kafka.MessageConsumer
-	DatasetAPIURL       string
-	DatasetAPIAuthToken string
-	MaxRetries          int
-	Producer            kafka.MessageProducer
-	S3                  *s3.S3
+	EnvMax                int64
+	BindAddr              string
+	Consumer              *kafka.ConsumerGroup
+	DatasetAPIURL         string
+	DatasetAPIAuthToken   string
+	DimensionExtractorURL string
+	HTTPClient            *rchttp.Client
+	MaxRetries            int
+	Producer              kafka.Producer
+	S3                    *s3.S3
+	Shutdown              time.Duration
 }
 
 // Start handles consumption of events
 func (svc *Service) Start() {
 	log.Info("application started", log.Data{"dataset_api_url": svc.DatasetAPIURL})
 
 	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, os.Kill)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	done := make(chan bool)
```

Comment:
can `done` be renamed to something more descriptive?  perhaps `serviceLoopDone`?